src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java

```diff
@@ -294,9 +307,39 @@ private List<OperatorStrategyResult> runActionSimulations(DcSecurityAnalysisCont
                 }
             }
         }
+
+        completeConnectivityResults(context, lfNetwork, propagatedContingencies, networkState);
+
         return operatorStrategyResults;
     }
 
+    private void completeConnectivityResults(DcSecurityAnalysisContext context, LfNetwork lfNetwork,
+                                             List<PropagatedContingency> propagatedContingencies,
+                                             NetworkState networkState) {
+
+        // some connectivity results have not been built yet and we have to.
+        // after some contingencies, no operator strategies have been applied.
+        // we apply the contingency on the network just to complete the connectivity results.
+        Iterator<PropagatedContingency> contingencyIt = propagatedContingencies.iterator();
+        while (contingencyIt.hasNext() && !Thread.currentThread().isInterrupted()) {
+            PropagatedContingency propagatedContingency = contingencyIt.next();
+            context.getConnectivityResultPerContingencyId().computeIfAbsent(propagatedContingency.getContingency().getId(), id ->
+                    propagatedContingency.toLfContingency(lfNetwork)
+                            .map(lfContingency -> {
+                                lfContingency.apply(context.getParameters().getLoadFlowParameters().getBalanceType());
+                                // we build the connectivity result linked to this contingency by opportunity.
+                                ConnectivityResult connectivityResult = new ConnectivityResult(lfContingency.getCreatedSynchronousComponentsCount(), 0,
+                                        lfContingency.getDisconnectedLoadActivePower() * PerUnit.SB,
+                                        lfContingency.getDisconnectedGenerationActivePower() * PerUnit.SB,
+                                        lfContingency.getDisconnectedElementIds());
+                                networkState.restore();
+                                return connectivityResult;
+                            })
+                            .orElse(new ConnectivityResult(0, 0, 0, 0, Collections.emptySet()))
```

Comment:
We have a API design issue, maybe we should have done this ConnectivityResult optional.

Reply:
If you have a `PostContingencyResult`, you have a `ConnectivityResult`, no?

Reply:
Bu if you don't have any loss of connectivity for you contingency, you should not provider this object?

Reply:
No... right...